istioctl/cmd/workload.go

```diff
@@ -0,0 +1,402 @@
+	return configureCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, outputDir string) error {
+	if err := validateWorkloadGroup(wg); err != nil {
+		return err
+	}
+	if err := os.MkdirAll(outputDir, filePerms); err != nil {
+		return err
+	}
+	if err := createClusterEnv(wg, outputDir); err != nil {
+		return err
+	}
+	if err := createCertsTokens(kubeClient, wg, outputDir); err != nil {
+		return err
+	}
+	meshConfig, err := createMeshConfig(kubeClient, wg, clusterID, revision, outputDir)
+	if err != nil {
+		return err
+	}
+	if err := createHosts(kubeClient, wg, meshConfig, outputDir); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Used to populate nil WorkloadGroup structs (metadata and template)
+// Additionally checks the invalid fields in template
+func validateWorkloadGroup(wg *clientv1alpha3.WorkloadGroup) error {
+	if wg.Spec.Metadata == nil {
+		wg.Spec.Metadata = &networkingv1alpha3.WorkloadGroup_ObjectMeta{}
+	}
+	if wg.Spec.Template == nil {
+		wg.Spec.Template = &networkingv1alpha3.WorkloadEntry{}
+	}
+	// error checks and warnings
+	template := wg.Spec.Template
+	if template.Address != "" {
+		return fmt.Errorf("address %s should not be set in the WorkloadEntry template", template.Address)
+	}
+	if len(template.Labels) != 0 {
+		fmt.Printf("Labels should be set in the metadata. The following WorkloadEntry labels will override metadata labels: %s", template.Labels)
+	}
+	// default service account for an empty field is "default"
+	if template.ServiceAccount == "" {
+		template.ServiceAccount = "default"
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, dir string) error {
+	we := wg.Spec.Template
+	ports := []string{}
+	for _, v := range we.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// respect the inbound port annotation and capture all traffic if no inbound ports are set
+	portBehavior := "*"
+	if len(ports) > 0 {
+		portBehavior = strings.Join(ports, ",")
+	}
+
+	// default attributes and service name, namespace, ports, service account, service CIDR
+	clusterEnv := map[string]string{
+		"ISTIO_CP_AUTH":       "MUTUAL_TLS",
+		"ISTIO_INBOUND_PORTS": portBehavior,
+		"ISTIO_NAMESPACE":     wg.Namespace,
+		"ISTIO_PILOT_PORT":    "15012",
+		"ISTIO_SERVICE":       fmt.Sprintf("%s.%s", wg.Name, wg.Namespace),
+		"ISTIO_SERVICE_CIDR":  "*",
+		"SERVICE_ACCOUNT":     we.ServiceAccount,
+	}
+	return ioutil.WriteFile(filepath.Join(dir, "cluster.env"), []byte(mapToString(clusterEnv)), filePerms)
```

Comment:
I think there are really two files in the original design: cluster.env, which has common config between all VMs, and sidecar.env, which is per VM config. Semantically they are the same though, so not a huge issue